src/containers/orgs/components/OrgRolesSection.js

```diff
@@ -136,22 +148,38 @@ class OrgRolesSection extends Component<Props, State> {
   closeModal = () => {
 
     const { actions, requestStates } = this.props;
+    const { isVisibleAddRoleModal } = this.state;
 
-    if (requestStates[ADD_ROLE_TO_ORGANIZATION] === RequestStates.STANDBY
-        || requestStates[REMOVE_ROLE_FROM_ORGANIZATION] === RequestStates.STANDBY) {
-      this.setState({
-        isVisibleAddRoleModal: false,
-        isVisibleRemoveRoleModal: false,
-      });
+    if (isVisibleAddRoleModal) {
+      /*
+       * the "add" modal needs special handling. we want to keep "valueOfRoleTitle" if the modal was simply just
+       * closed, i.e. the add action was not taken. we want to clear "valueOfRoleTitle" only when the add action
+       * was actually taken
+       */
+      if (requestStates[ADD_ROLE_TO_ORGANIZATION] === RequestStates.STANDBY) {
+        // clicking "cancel" instead of "confirm", i.e. closing the modal without taking action
+        this.setState({
+          isVisibleAddRoleModal: false,
+        });
+      }
+      else {
+        // clicking "close" after having clicked "confirm", i.e. closing the modal after taking action
+        this.setState({
+          isValidRole: true,
+          isVisibleAddRoleModal: false,
+          selectedRoleId: undefined,
+          valueOfRoleTitle: '',
+        });
+        setTimeout(() => {
+          actions.resetRequestState(ADD_ROLE_TO_ORGANIZATION);
+        }, 1000);
```

Comment:
Can you explain the timeout? If you want to guarantee this occurs after the setState(), you can provide a callback to setState as a second param.

```this.setState({...}, () => actions.resetRequestState(...));```

Reply:
I forgot to copy over the timeout comment from below; updated. The `setState()` callback happens too quickly, i.e. the modal transition animation doesn't have enough time to finish.

Reply:
Oh I see. Maybe we should expose the callbacks on the underlying transition itself? Something like this might be defeated if someone reopens the same modal before the timeout occurs right?
https://reactcommunity.org/react-transition-group/css-transition#CSSTransition-prop-onExited